utilities/pipelines/resourceRemoval/helper/Invoke-ResourcePostRemoval.ps1

```diff
@@ -149,6 +149,20 @@ function Invoke-ResourcePostRemoval {
             $null = Set-AzRecoveryServicesVaultProperty -VaultId $vaultId -SoftDeleteFeatureState $softDeleteStatus.TrimEnd('d')
             break
         }
+        'Microsoft.MachineLearningServices/workspaces' {
```

Comment:
I guess this is ideally not necessary if the main removal (& purge) worked. That said, I guess we would need either a try catch or a GET first in case we try to remove a already purged workspace, right?

Reply:
Seems it marks it for deletion but goes on. Deletion takes a while sometimes, but not sure that is why it is failing. Also seems another part of the removal function doesn't catch the lock on the common test.

Reply:
Regarding the lock it would be odd as we do this everywhere. However, it is indeed not 100% reliable as it can take longer than the period we wait to delete the lock (in rare cases). Happy the merge the change in, as long as we don't introduce a new issue in case a ML workspace was already purged.

Reply:
Think its submitting a deletion, carrying on and it gets deleted as part of another delete (RG delete maybe). What happens is it ends up being soft deleted. Could add a loop on the lock to see if the lock still exists before carrying on. I.e. a hold for a minute or so. Exploring this as we speak.

Reply:
Tried that and it didn't work. The lock shows up as being deleted even though it's still blocked somewhere. The best we were able to come up with was the lock wait

Reply:
Also testing a loop on the ML workspace.

Reply:
I see the latest API allowing forceandpurge is now `2023-06-01-preview`, which compared to the `2023-04-01-preview` may have the advantage of not being an April 1st version 😆 But seriously, we may at least try if more stable

Reply:
Doing that on my local/hidden branch. Also a small refresh on logging.